### ISSUE TITLE
Manjaro or Arch

### ISSUE THREAD

**Issue by massatt212** [Samsagax/gamescope-session#2]
I have Gnome install which is GDM, if i disable GDM and eable gamescope session i get a black screen, how do i bypass that?

**Comment by Samsagax** [comment]
Works for me. What hardware are you testing on?

**Comment by massatt212** [comment]
i got it working with steamdeck ui, by changing the user name on your gamescope@tty1 session.

**Comment by massatt212** [comment]
Nice, can't wait to test it out, is their a way to change gamescope default resolution to 1080p?

**Comment by massatt212** [comment]
Instead of disabling lightdm sddm or gdm, is their a way to keep them but when selecting desktop environment it changes to gamescope compositor? Cause it would be much better to have my login screen instead of enabling and disabling all the time, or it can't be done?

**Comment by Samsagax** [comment]
> Nice, can't wait to test it out, is their a way to change gamescope default resolution to 1080p?

You can change the resolution by setting environment variables accordingly in a `*.conf` file under `~/.config/environment.d/`
```
# Size of the screen
SCREEN_H=2160
SCREEN_W=3840

# Size to render
RENDER_H=2160
RENDER_W=3840
```

[This are the defaults](https://github.com/Samsagax/gamescope-session/blob/main/etc/gamescope-session.conf)

As for the session under sddm or any other session manager it is possible, but this is meant to be a standalone session for a console-like setup. You could add the relevant file that invokes the session script to accommodate for your favorite session manager as it is agnostic. I don't mind adding it to the repo

**Comment by massatt212** [comment]
Nice, can't wait to test it out, is their a way to change gamescope default resolution to 1080p?


Please do, it will feel like steam os when we could of go to desktop mode, u can do both and see which brings good results.

**Comment by Samsagax** [comment]
This repo is not meant to be a session outside of what a console needs so I don't have plans on supporting session managers right now.

Anyway, closing the issue as the main problem seems resolved.

**Comment by massatt212** [comment]
I'll check it out

**Comment by massatt212** [comment]
Just as suspected, mouse does not work properly, try out apex legends, and see for your self, but nice work so far, enter -steampal to have access to mangohud, and see if you can get -steamos3 working also.

**Comment by massatt212** [comment]
#!/bin/sh

# Source global variables
. /etc/gamescope-session.conf

# Some environment variables by default (taken from Deck session)
export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0
export STEAM_USE_MANGOAPP=1
export R600_DEBUG=nodcc

# Enable volume key management via steam for this session
export STEAM_ENABLE_VOLUME_HANDLER=1

# Have SteamRT's xdg-open send http:// and https:// URLs to Steam
export SRT_URLOPEN_PREFER_STEAM=1

# Disable automatic audio device switching in steam, now handled by wireplumber
export STEAM_DISABLE_AUDIO_DEVICE_SWITCHING=1

# Enable support for xwayland isolation per-game in Steam
export STEAM_MULTIPLE_XWAYLANDS=1

# Don't wait for buffers to idle on the client side before sending them to gamescope
export vk_xwayland_wait_ready=false

# Source environment from ~/.config/environment.d
set -a
for i in "${HOME}"/.config/environment.d/*.conf ;
do
    [[ -f "${i}" ]] && . "${i}"
done
set +a

# Setup socket for gamescope
# Create run directory file for startup and stats sockets
tmpdir="$([[ -n ${XDG_RUNTIME_DIR+x} ]] && mktemp -p "$XDG_RUNTIME_DIR" -d -t gamescope.XXXXXXX)"
socket="${tmpdir:+$tmpdir/startup.socket}"
# Fail early if we don't have a proper runtime directory setup
if [[ -z $tmpdir || -z ${XDG_RUNTIME_DIR+x} ]]; then
	echo >&2 "!! Failed to find run directory in which to create stats session sockets (is \$XDG_RUNTIME_DIR set?)"
	exit 0
fi

mkfifo -- "$socket"

# Use cursor if file exist
if [ -z "$CURSOR_FILE" ] ; then
    CURSOR_FILE="${HOME}/.local/share/Steam/tenfoot/resource/images/cursors/arrow_right.png"
fi
CURSOR=""
if [ -f "$CURSOR_FILE" ] ; then
    CURSOR="--cursor ${CURSOR_FILE}"
fi

# Define session if not overriden
if [ -z "$STEAMCMD" ] ; then
    STEAMCMD="steam -gamepadui -steamos -steampal -pipewire-dmabuf"
    # Check beta branch and select prefered UI
    STEAM_BETA_FILE="${HOME}/.local/share/Steam/package/beta"
    if [ -f "$STEAM_BETA_FILE" ] ; then
        BETA=$(cat "${STEAM_BETA_FILE}")
        if [[ $BETA =~ steampal ]] ; then
            STEAMCMD="steam -gamepadui -steamos -steampal -pipewire-dmabuf"
        fi
    fi
fi
if [ -z "$GAMESCOPECMD" ] ; then
    GAMESCOPECMD="/usr/bin/gamescope \
      $CURSOR \
      -U \
      -e \
      -R "$socket" \
      -W $SCREEN_W -H $SCREEN_H \
      -w $RENDER_W -h $RENDER_H \
      --xwayland-count 2"
fi

# Workaround for Steam login issue while Steam client change propagates out of Beta
touch "${HOME}"/.steam/root/config/SteamAppData.vdf || true

# Log rotate the last session
if [ -f "${HOME}"/.steam-tweaks.log ]; then
    cp "${HOME}"/.steam-tweaks.log "${HOME}"/.steam-tweaks.log.old
fi
if [ -f "${HOME}"/.steam-stdout.log ]; then
    cp "${HOME}"/.steam-stdout.log "${HOME}"/.steam-stdout.log.old
fi
if [ -f "${HOME}"/.gamescope-stdout.log ]; then
    cp "${HOME}"/.gamescope-stdout.log "${HOME}"/.gamescope-stdout.log.old
fi

# Run steam-tweaks if exists
if command -v steam-tweaks > /dev/null; then
    steam-tweaks > "${HOME}"/.steam-tweaks.log
fi

# Start gamescope compositor, log it's output and background it
$GAMESCOPECMD > "${HOME}"/.gamescope-stdout.log 2>&1 &
gamescope_pid="$!"

if read -r -t 3 response_x_display response_wl_display <> "$socket"; then
	export DISPLAY="$response_x_display"
	export GAMESCOPE_WAYLAND_DISPLAY="$response_wl_display"
	# We're done!
else
	kill -9 "$gamescope_pid"
	wait
	exit 0
	# Systemd or Session manager will have to restart session
fi

# If we have mangoapp binary start it
if command -v mangoapp > /dev/null; then
    mangoapp &
fi

# Start Steam client
$STEAMCMD > "${HOME}"/.steam-stdout.log 2>&1

# When the client exits, kill gamescope nicely
kill $gamescope_pid

# Start a background sleep for five seconds because we don't trust it
sleep 5 &
sleep_pid="$!"

# Wait for gamescope or the sleep to finish for timeout purposes
ret=0
wait -n $gamescope_pid $sleep_pid || ret=$?

# If we get a SIGTERM/etc while waiting this happens.  Proceed to kill -9 everything but complain
if [[ $ret = 127 ]]; then
	echo >&2 "gamescope-session: Interrupted while waiting on teardown, force-killing remaining tasks"
fi

# Kill all remaining jobs and warn if unexpected things are in there (should be just sleep_pid, unless gamescope failed
# to exit in time or we hit the interrupt case above)
for job in $(jobs -p); do
	# Warn about unexpected things
	if [[ $ret != 127 && $job = "$gamescope_pid" ]]; then
		echo >&2 "gamescope-session: gamescope timed out while exiting, killing"
	elif [[ $ret != 127 && $job != "$sleep_pid" ]]; then
		echo >&2 "gamescope-session: unexpected background pid $job at teardown: "
		# spew some debug about it
		ps -p "$job" >&2
	fi
	kill -9 "$job"
done

**Comment by massatt212** [comment]
[steamos-session.zip](https://github.com/Samsagax/gamescope-session/files/8367047/steamos-session.zip)

This is what i use and get Mouse Working
but i cant get -steamdeck working
-steamos3 sometimes produce bad resolution

**Comment by Samsagax** [comment]
Both `-steamdeck` and `-steamos3` will break resolution (specially with the game overlay). But good to know about `-steampal` option. Will try it.

**Comment by massatt212** [comment]
R600_DEBUG=nodcc how to I enable this in your option.

**Comment by Samsagax** [comment]
> R600_DEBUG=nodcc how to I enable this in your option.

This discussion is getting off-topic with the issue. You could open a new issue or go to the ChimeraOS discord (we are official now :P) with this kinds of questions.

But for the answer: anything you put inside a `.conf` file under `~/.config/environment.d/` will be sourced by the session. So you could place it there.
